### PR TITLE
feat(discover): Support release:latest in discover

### PR DESCRIPTION
**Pull request by wmak**
- Adds support for `release:latest` in discover
    - Changed `environment` in `get_filter_params` to be the environment objects
    - Adding organization_id as a default there as well.
- Adds `latest` as an autocomplete option in the searchbar

### CODE REVIEW

File: src/sentry/api/bases/organization.py [hunk 1]
```diff
@@ -276,8 +276,13 @@ def get_filter_params(self, request, organization, date_filter_optional=False):
         if not projects:
             raise NoProjects
 
-        environments = [env.name for env in self.get_environments(request, organization)]
-        params = {"start": start, "end": end, "project_id": [p.id for p in projects]}
+        environments = self.get_environments(request, organization)
```
Comment (markstory) [inline src/sentry/api/bases/organization.py:279]: This seems like a risky change. Couldn't this be used in many endpoints?

File: src/sentry/search/utils.py [hunk 2]
```diff
@@ -224,10 +224,11 @@ def parse_user_value(value, user):
         return User(id=0)
 
 
-def get_latest_release(projects, environments):
-    release_qs = Release.objects.filter(
-        organization_id=projects[0].organization_id, projects__in=projects
-    )
+def get_latest_release(projects, environments, organization_id=None):
+    if organization_id is None:
+        organization_id = projects[0].organization_id
```
Comment (markstory) [inline src/sentry/search/utils.py:229]: When would you potentially want a different organization_id than what is on the project?

Comment (wmak) [reply src/sentry/search/utils.py:229]: In this case its because we only have a list of project_ids but still have access to the organization_id

File: src/sentry/api/bases/organization.py (hunk 1 above)
Comment (wmak) [reply src/sentry/api/bases/organization.py:279]: oof yeah, this breaks a lot of things. I'll try to find a less risky way to do this :+1:

File: src/sentry/search/utils.py (hunk 2 above)
Comment (markstory) [reply src/sentry/search/utils.py:229]: Should we update the other usages of this to also include the organization_id?

**Review by markstory — APPROVED** [review]

File: src/sentry/search/utils.py (hunk 2 above)
Comment (wmak) [reply src/sentry/search/utils.py:229]: Other usage of this is in, `GroupEventsEndpoint` which gets the org even for other functions via `group.project.organization`

File: src/sentry/api/event_search.py [hunk 3]
```diff
@@ -883,7 +902,7 @@ def to_list(value):
             else:
                 kwargs["project_ids"] = params["project_id"]
         if "environment" in params:
-            term = SearchFilter(SearchKey("environment"), "=", SearchValue(params["environment"]))
+            term = SearchFilter(SearchKey("environment"), "=", SearchValue(params["environment"]),)
```
Comment (evanh) [inline src/sentry/api/event_search.py:905]:
```suggestion
            term = SearchFilter(SearchKey("environment"), "=", SearchValue(params["environment"]))
```

File: src/sentry/api/endpoints/organization_events.py [hunk 4]
```diff
@@ -126,8 +126,6 @@ def get(self, request, organization):
             except NoProjects:
                 return Response([])
 
-            params["organization_id"] = organization.id
```
Comment (evanh) [inline src/sentry/api/endpoints/organization_events.py:129]: Why did you remove this? It seems like you're using it when you call `parse_release`

File: src/sentry/api/event_search.py [hunk 5]
```diff
@@ -853,6 +855,23 @@ def to_list(value):
...
                     kwargs["conditions"].append(user_conditions)
+            elif name == RELEASE_ALIAS:
+                converted_filter = convert_search_filter_to_snuba_query(
+                    SearchFilter(
+                        term.key,
+                        term.operator,
+                        SearchValue(
+                            parse_release(
+                                term.value.value,
+                                params["project_id"],
```
Comment (evanh) [inline src/sentry/api/event_search.py:866]: This seems like it might break, since `get_latest_release` expects actual project objects, not IDs (`organization_id = projects[0].organization_id`)

File: src/sentry/api/endpoints/organization_events.py (hunk 4 above)
Comment (wmak) [reply src/sentry/api/endpoints/organization_events.py:129]: Aiming for consistency where we set stuff on the `params` dict, I moved this into the `get_filter_params` [here](https://github.com/getsentry/sentry/pull/18568/files#diff-f348b3a3d06653db5b246ec567d39dc4R286)

File: src/sentry/api/event_search.py (hunk 5 above)
Comment (wmak) [reply src/sentry/api/event_search.py:866]: Poor naming in `get_filter_params` `project_id` is a list of objects, not ids see [here](https://github.com/getsentry/sentry/pull/18568/files#diff-f348b3a3d06653db5b246ec567d39dc4L282)

Worth changing in another PR I think though since that was my first impression of that too

Comment (wmak) [reply src/sentry/api/event_search.py:866]: Sorry previous comment was completely incorrect, this is more a django thing
```
Release.objects.filter(project__in=[1,2,3])
```

```
projects = [p for p in Project.objects.filter(id__in=[1,2,3])]
Release.objects.filter(project__in=projects)
```

Become the same query with `...WHERE "sentry_release_project"."project_id" IN (1, 2, 3)`

Comment (evanh) [reply src/sentry/api/event_search.py:865]: Won't this line fail: https://github.com/getsentry/sentry/pull/18568/files#diff-712e897580b290560333dc7b4bb0dd88R229

Since an int does not have an attribute 'organization_id'.

Comment (wmak) [reply src/sentry/api/event_search.py:865]: Hmm I'll add a check there, but in this case the organization_id is passed from `params` so the function will use that instead of trying to pull it off the projects.

The function is shared with `GroupEventsEndpoint` which only has the list of project objects, but no org which is why it has to grab the org_id

File: src/sentry/api/event_search.py [hunk 6]
```diff
@@ -829,30 +830,47 @@ def to_list(value):
...
+                converted_filter = convert_search_filter_to_snuba_query(
+                    SearchFilter(
+                        term.key,
+                        term.operator,
+                        SearchValue(
+                            parse_release(
+                                value,
+                                params["project_id"],
+                                params.get("environment_objects"),
+                                params["organization_id"],
```
Comment (wedamija) [inline src/sentry/api/event_search.py:867]: @wmak `organization_id` isn't always in params here, and causes this error: https://sentry.io/organizations/sentry/issues/1846529501/?project=1&query=is%3Aunresolved+assigned%3A%23workflow&statsPeriod=14d. Mind taking a look?

Comment (wmak) [reply src/sentry/api/event_search.py:867]: :+1: can do thanks!

Comment (wmak) [reply src/sentry/api/event_search.py:867]: Thanks again for the heads up, fix here: https://github.com/getsentry/sentry/pull/20790
